test/rules/unrelated_type_equality_checks_test.dart

```diff
@@ -18,6 +18,28 @@ class UnrelatedTypeEqualityChecksTestLanguage300 extends LintRuleTest
   @override
   String get lintRule => 'unrelated_type_equality_checks';
 
+  test_recordAndInterfaceType_unrelated() async {
```

Comment:
It would be good to have some tests involving records with named fields, and maybe some that have both positional and named fields.

Reply:
Done!